Sources/SwiftSyntax/SyntaxNodes.swift.gyb.template

```diff
@@ -194,7 +194,14 @@ public struct ${node.name}: ${base_type}Protocol, SyntaxHashable {
       return ${child_type}(childData!)
     }
     set(value) {
-      self = with${child.name}(value)
+      let arena = SyntaxArena()
+  %       if child.is_optional:
+      let raw = value?.raw
+  %       else:
+      let raw = value.raw
+  %       end
+      let newData = data.replacingChild(at: ${idx}, with: raw, arena: arena)
+      self = ${node.name}(newData)
```

Comment:
Ah, I was hoping we could use `SyntaxNodeStructure` and the need for this entirely. We don't currently have the reverse mapping (ie. keypath -> index), but maybe that would be useful. That would be even more codegen though, not less 😅.

Reply:
How would you want to get rid of these using `SyntaxNodeStructure`?

Reply:
You can't right, we need the reverse mapping (key path -> index).